bindings/wasm/src/account/wasm_account.rs

```diff
@@ -0,0 +1,365 @@
+        .as_ref()
+        .borrow_mut()
+        .sign(fragment.as_str(), &mut pres, options)
+        .await
+        .map(|_| JsValue::undefined())
+        .wasm_result()?;
+      JsValue::from_serde(&pres).wasm_result()
+    });
+    promise.unchecked_into::<PromisePresentation>()
+  }
+
+  /// Signs arbitrary `data` with the key specified by `fragment`.
+  #[wasm_bindgen(js_name = createSignedData)]
+  pub fn create_signed_data(
+    &self,
+    fragment: String,
+    data: &JsValue,
+    signature_options: WasmSignatureOptions,
+  ) -> Result<Promise> {
+    let account = self.0.clone();
+    let mut verifiable_properties: VerifiableProperties = data.into_serde().wasm_result()?;
+    let options: SignatureOptions = SignatureOptions::from(signature_options);
+
+    let promise = future_to_promise(async move {
+      account
+        .as_ref()
+        .borrow_mut()
+        .sign(fragment.as_str(), &mut verifiable_properties, options)
+        .await
+        .map(|_| JsValue::undefined())
+        .wasm_result()?;
+      JsValue::from_serde(&verifiable_properties).wasm_result()
+    });
+    Ok(promise)
+  }
+
+  /// Overwrites the {@link Document} this account manages, **without doing any validation**.
+  ///
+  /// # WARNING
+  ///
+  /// This method is dangerous and can easily corrupt the internal state,
+  /// potentially making the identity unusable. Only call this if you fully
+  /// understand the implications!
+  #[wasm_bindgen(js_name = updateDocumentUnchecked)]
+  pub fn update_document_unchecked(&mut self, document: WasmDocument) -> Promise {
+    let account = self.0.clone();
+    future_to_promise(async move {
+      account
+        .as_ref()
+        .borrow_mut()
+        .update_document_unchecked(document.0)
+        .await
+        .map(|_| JsValue::undefined())
+        .wasm_result()
+    })
+  }
+
+  /// Fetches the latest changes from the tangle and **overwrites** the local document.
+  ///
+  /// If a DID is managed from distributed accounts, this should be called before making changes
+  /// to the identity, to avoid publishing updates that would be ignored.
+  #[wasm_bindgen(js_name = fetchState)]
+  pub fn fetch_state(&mut self) -> Promise {
+    let account = self.0.clone();
+    future_to_promise(async move {
+      account
+        .as_ref()
+        .borrow_mut()
+        .fetch_state()
+        .await
+        .map(|_| JsValue::undefined())
+        .wasm_result()
+    })
+  }
+}
+
+impl From<Account> for WasmAccount {
+  fn from(account: Account) -> WasmAccount {
+    WasmAccount(Rc::new(WasmRefCell::new(account)))
+  }
+}
+
+#[wasm_bindgen]
+extern "C" {
+  #[wasm_bindgen(typescript_type = "Promise<Credential>")]
+  pub type PromiseCredential;
+}
+
+#[wasm_bindgen]
+extern "C" {
+  #[wasm_bindgen(typescript_type = "Promise<Presentation>")]
+  pub type PromisePresentation;
+}
+
+#[wasm_bindgen]
+extern "C" {
+  #[wasm_bindgen(typescript_type = "Promise<Document>")]
+  pub type PromiseDocument;
+}
```

Comment:
Combine these `extern "C"` sections? Otherwise I wonder if these should all be moved to their respective Wasm types for code organisation.